packages/components/src/modal/src/components/osds-modal/osds-modal.tsx

```diff
@@ -87,7 +87,8 @@ export class OsdsModal implements OdsModalAttribute, OdsModalMethod, OdsModalEve
   }
 
   disconnectedCallback(): void {
-    this.close();
+    document.body.style.removeProperty('overflow');
+    this.masked = true;
```

Comment:
put those in a dedicated function and call it from here and close `method`

Reply:
Done